Windows - add a DEBUG build with debugging symbols

EdmondDantes opened the pull request:
This change enables the extension to be compiled correctly in debug mode.
The issue was that, in the default case, the builder attempted to use files from the PHP DEV PACK, which lacks debug symbols. This resulted in a linker error, as it couldn't find the emalloc functions (these have different signatures in DEBUG builds).

Solution:
* Add an environment variable `PHP_LIB`.
* Check if we are compiling in debug mode. If so, require this variable to be set. Otherwise, proceed as before.
* `PHP_LIB` must point to a built `PHP` instance. The library name is chosen as `php8ts_debug.lib`.

P.S. I tried to find a place in the documentation to add this, but I can't figure out where it should go. If you can provide details, I would certainly include it.

Xenira commented:
Thanks for the contribution and sorry for the delay in getting back to you.

Regarding the documentation it would probably be best to add a separate "Windows" section here as I could not find a good place to put it.

Seems like the pipeline builds are now unable to find the proper php lib. Can you have a look at that?

EdmondDantes commented:
@Xenira Thank you for the suggestions.

I fixed the library search function to ensure it works in all cases or returns an error.
I added a brief note that for DEBUG builds on Windows, PHP must be manually built from the SDK.

Xenira commented on windows_build.rs:
```diff
@@ -233,13 +233,42 @@ impl DevelPack {
+            .iter()
+            .map(|lib| php_lib_path.join(lib))
+            .find(|path| path.exists())
+            .unwrap_or_else(||
+                panic!(
+                    "{}",
+                    if is_debug {
+                        format!(
+                            r#"Error: No suitable PHP library found in '{}'.
+To build the application in DEBUG mode on Windows,
+you must have a PHP SDK built with the DEBUG option enabled
+and specify the PHP_LIB to the folder containing the lib files.
+For example: set PHP_LIB=C:\php-sdk\php-dev\vc16\x64\php-8.3.13-src\x64\Debug_TS."#,
+                            php_lib_path.display()
+                        )
+                    } else {
+                        format!("Error: No suitable PHP library found in '{}'.", php_lib_path.display())
+                    }
+                ))
```
use `expect` here?

EdmondDantes commented:
@Xenira Thanks!

Xenira commented:
@EdmondDantes could you run `cargo fmt` on your changes?

EdmondDantes commented:
@Xenira Done

Xenira commented:
Updated the branch with clippy fixes from master #352
